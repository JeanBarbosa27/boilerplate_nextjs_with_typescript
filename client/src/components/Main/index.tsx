import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Image de um átomo e React Avançado ao lado."
    />
    <S.Title>React Avançado - Boilerplate</S.Title>
    <S.Description>
      TypeScript, ReactJS, NextJS e Styeld Components!
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Pessoa desenvolvendo código de programação no computador"
    />
  </S.Wrapper>
)

export default Main
