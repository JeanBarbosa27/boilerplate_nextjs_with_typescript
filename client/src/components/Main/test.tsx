import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should be able to render Main componente', () => {
    const { container } = render(<Main />)
    const heading = screen.getByRole('heading', { name: /react avan√ßado/i })

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
