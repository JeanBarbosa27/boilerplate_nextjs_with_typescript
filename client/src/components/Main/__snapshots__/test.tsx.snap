// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be able to render Main componente 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`
